src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = calculateIndex(key);
+        Node<K, V> firstNode = table[index];
+        Node<K, V> newNode = new Node<>(key, value);
+        if (firstNode == null) {
+            table[index] = newNode;
+            size++;
+            resizeIfNeeded();
+            return;
+        }
+        addOrReplace(firstNode, newNode);
     }
 
     @Override
     public V getValue(K key) {
+        int index = calculateIndex(key);
+        Node<K, V> firstNode = table[index];
+        for (Node<K, V> node = firstNode; node != null; node = node.next) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void addOrReplace(Node<K, V> firstNode, Node<K, V> newNode) {
+        Node<K, V> node = new Node<>(null, null);
+        node.next = firstNode;
+        do {
+            node = node.next;
+            if (Objects.equals(node.key, newNode.key)) {
+                node.value = newNode.value;
+                return;
+            }
+        } while (node.next != null);
+        node.next = newNode;
+        size++;
+        resizeIfNeeded();
+    }
+
+    private void resizeIfNeeded() {
+        if (size > threshold()) {
+            grow();
+        }
+    }
+
+    private void grow() {
+        int capacity = table.length << 1;
+        Node<K, V>[] oldTable = table;
+        size = 0;
+        table = new Node[capacity];
+        transfer(oldTable);
+    }
+
+    private void transfer(Node<K, V>[] oldTable) {
+        for (Node<K, V> bucket : oldTable) {
+            while (bucket != null) {
+                put(bucket.key, bucket.value);
+                bucket = bucket.next;
+            }
+        }
+    }
+
+    private int calculateIndex(K key) {
+        return (key == null ? 0 : key.hashCode()) & (table.length - 1);
+    }
+
+    private float threshold() {
```

Comment:
```suggestion
    private float getThreshold() {
```
method names should be verbs